src/test/java/com/fittracker/fittracker/controller/WeightControllerValidationTest.java

```diff
@@ -0,0 +1,114 @@
+                of(ENDPOINT, BAD_REQUEST, "{\"field\":\"date\",\"message\":\"Parameter must not be null\"}", never()),
+                of(String.format(PARAM_ENDPOINT, "?date=123"), BAD_REQUEST, "{\"field\":\"date\",\"message\":\"Invalid data type\"}", never()),
+                of(String.format(PARAM_ENDPOINT, "?date=2020-01-01"), OK, "", times(1)),
+                of(String.format(PARAM_ENDPOINT, "?date=2050-10-10"), OK, "", times(1))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getRequestTestDataProvider")
+    void givenGetUriRequest_shouldReturnResponse(String uri, HttpStatus responseStatus, String responseBody, VerificationMode numberOfServiceInvocations) throws Exception {
+        mockMvc
+                .perform(get(URI.create(uri)))
+                .andExpect(status().is(responseStatus.value()))
+                .andExpect(content().string(responseBody));
+
+        verify(weightService, numberOfServiceInvocations).findByDate(any());
+    }
+
+    private static Stream<Arguments> postRequestTestDataProvider() {
+
+        return Stream.of(
+                of(ENDPOINT, CREATED, "{\"date\":\"2023-08-03\",\"value\":13.2}", "", times(1)),
+                of(ENDPOINT, CREATED, "{\"date\":\"" + LocalDate.now() + "\",\"value\":13.2}", "", times(1)),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"202-08-03\",\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date must be in format: YYYY-MM-DD\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"03-08-2023\",\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date must be in format: YYYY-MM-DD\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date must not be null\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"" + randomDateInTheFuture() + "\",\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date cannot be in the future\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"2021-01-02\",\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date must be after 2022\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"2022-12-31\",\"value\":13.2}",
+                        "{\"field\":\"date\",\"message\":\"Date must be after 2022\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"2023-08-03\"}",
+                        "{\"field\":\"value\",\"message\":\"Value must not be null\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"2023-08-03\",\"value\":-1}",
+                        "{\"field\":\"value\",\"message\":\"Value must be positive\"}", never()),
+                of(ENDPOINT, BAD_REQUEST, "{\"date\":\"2023-08-03\",\"value\":700}",
+                        "{\"field\":\"value\",\"message\":\"Value must be less than 635\"}", never())
+                );
```

Comment:
It's really difficult to read it, I have a few ideas on how we can improve in that matter, let's have a chat about it.